packages/loopring_v3/contracts/core/impl/libtransactions/SignatureVerificationTransaction.sol

```diff
@@ -26,20 +26,24 @@ library SignatureVerificationTransaction
 
     function readTx(
         bytes memory data,
-        uint         offset
+        uint         offset,
+        SignatureVerification memory verification
         )
         internal
         pure
-        returns (SignatureVerification memory verification)
     {
         uint _offset = offset;
+
+        require(data.toUint8Unsafe(_offset) == uint8(ExchangeData.TransactionType.SIGNATURE_VERIFICATION), "INVALID_TX_TYPE");
```

Comment:
These functions are now used directly in the AMM pool contracts as well, so instead of creating extra functions that just do this additional check I just added these checks here because it is cheap and also be a bit cleaner I think (complete transaction data is now read here instead of jumping 1 byte ahead every time).